src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.IncrementalAnalyzer.AnalyzerExecutor.cs

```diff
@@ -169,16 +169,19 @@ public AnalyzerExecutor(DiagnosticIncrementalAnalyzer owner)
                         continue;
                     }
 
+                    textVersion = existingData.TextVersion;
                     dataVersion = existingData.DataVersion;
+
                     builder.AddRange(existingData.Items);
                 }
 
                 if (dataVersion == VersionStamp.Default)
                 {
+                    Contract.Requires(textVersion == VersionStamp.Default);
                     return null;
                 }
 
-                return new AnalysisData(VersionStamp.Default, dataVersion, builder.ToImmutable());
+                return new AnalysisData(textVersion, dataVersion, builder.ToImmutable());
```

Comment:
Minor: Assert text version is not default here or probably in ctor of AnalysisData?